src/xbot/common/properties/PermanentStorageBase.java

```diff
@@ -1,24 +1,80 @@
 package xbot.common.properties;
 
 import java.sql.Connection;
+import java.sql.DatabaseMetaData;
 import java.sql.DriverManager;
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.sql.Statement;
+
 import org.apache.log4j.Logger;
 
 /**
  *
  * @author John
  */
-public class PermanentStorage extends PermanentStorageProxy {
+public class PermanentStorageBase extends PermanentStorageProxy {
```

Comment:
I think we may keep the two levels for now, but if we find ourselves not using the other Mock, collapse them.

Abstract is done.